src/view/styles.js

```diff
@@ -0,0 +1,490 @@
+	 * or many styles at once:
+	 *
+	 *		styles.insertProperty( {
+	 *			color: 'blue',
+	 *			'margin-right': '1em'
+	 *		} );
+	 *
+	 * Supports shorthands.
+	 *
+	 * @param {String|Object} nameOrObject
+	 * @param {String|Object} value
+	 * @returns {Boolean}
+	 */
+	insertProperty( nameOrObject, value ) {
+		if ( isObject( nameOrObject ) ) {
+			for ( const key of Object.keys( nameOrObject ) ) {
+				this.insertProperty( key, nameOrObject[ key ] );
+			}
+		} else {
+			this.converter.toNormalizedForm( nameOrObject, value, this._styles );
+		}
+	}
+
+	/**
+	 * Removes styles property.
+	 *
+	 * @param name
+	 */
+	removeProperty( name ) {
+		unset( this._styles, toPath( name ) );
+		delete this._styles[ name ];
+	}
+
+	/**
+	 * Return normalized style object;
+	 *
+	 *		const styles = new Styles();
+	 *		styles.setStyle( 'margin:1px 2px 3em;' );
+	 *
+	 *		console.log( styles.getNormalized( 'margin' ) );
+	 *		// will log:
+	 *		// {
+	 *		//     top: '1px',
+	 *		//     right: '2px',
+	 *		//     bottom: '3em',
+	 *		//     left: '2px'
+	 *		// }
+	 *
+	 * @param {String} name
+	 * @returns {Object|undefined}
+	 */
+	getNormalized( name ) {
+		return this.converter.getNormalized( name, this._styles );
+	}
+
+	/**
+	 * Returns a string containing normalized styles string or undefined if no style properties are set.
+	 *
+	 * @returns {String|undefined}
+	 */
+	getInlineStyle() {
+		const entries = this._getStylesEntries();
+
+		// Return undefined for empty styles map.
+		if ( !entries.length ) {
+			return;
+		}
+
+		return entries.map( arr => arr.join( ':' ) ).join( ';' ) + ';';
+	}
+
+	/**
+	 * Returns property value string.
+	 *
+	 * @param {String} propertyName
+	 * @returns {String|undefined}
+	 */
+	getInlineProperty( propertyName ) {
+		const normalized = this.converter.getNormalized( propertyName, this._styles );
+
+		if ( !normalized ) {
+			// Try return styles set directly - values that are not parsed.
+			return this._styles[ propertyName ];
+		}
+
+		if ( isObject( normalized ) ) {
+			const styles = this.converter.getReducedForm( propertyName, normalized );
+
+			const propertyDescriptor = styles.find( ( [ property ] ) => property === propertyName );
+
+			// Only return a value if it is set;
+			if ( Array.isArray( propertyDescriptor ) ) {
+				return propertyDescriptor[ 1 ];
+			}
+		} else {
+			return normalized;
+		}
+	}
+
+	/**
+	 * Returns style properties names as the would appear when using {@link #getInlineStyle}
+	 *
+	 * @returns {module:engine/view/styles~PropertyEntry}
+	 */
+	getStyleNames() {
+		const entries = this._getStylesEntries();
+
+		return entries.map( ( [ key ] ) => key );
+	}
+
+	/**
+	 * Removes all styles.
+	 */
+	clear() {
+		this._styles = {};
+	}
+
+	/**
+	 * Returns normalized styles entries for further processing.
+	 *
+	 * @private
+	 * @returns {module:engine/view/styles~PropertyEntry}
+	 */
+	_getStylesEntries() {
+		const parsed = [];
+
+		const keys = Object.keys( this._styles ).sort();
```

Comment:
I don't see a need for sorting here. Isn't sorting only needed for getting things as string?

Reply:
(also, performance)